network/src/connectivity_manager/mod.rs

```diff
@@ -205,6 +223,36 @@ impl DiscoveredPeer {
     pub fn is_eligible_to_be_dialed(&self) -> bool {
         self.is_eligible() && !self.addrs.is_empty()
     }
+
+    /// Updates the last time we tried to connect to this node
+    pub fn set_last_dial_time(&mut self, time: SystemTime) {
```

Comment:
nit: you could probably avoid passing in a system time if it's always going to be: `SystemTime::now()`

Reply:
This is so I can test with it in some tests later by not fixing the SystemTime